docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
-# Developer Guide
-
+# Property Rental Manager - Developer Guide
+* [Acknowledgements](#acknowledgements)
+* [Setting Up and Getting Started](#setting-up-and-getting-started)
+* [Product Scope](#product-scope)
+* [Target User Profile](#target-user-profile)
+* [Value Proposition](#value-proposition)
+* [User Stories](#user-stories)
+* [Design](#design)
+  * [Architecture](#architecture)
+  * [UI Component](#ui-component)
+  * [Client Component](#client-component)
+  * [Property Component](#property-component)
+  * [List Component](#list-component)
+  * [Model Component](#model-component)
+  * [List Component](#list-component)
+  * [Storage Component](#storage-component)
+  * [Common Classes](#storage-component)
+* [Implementation](#implementation)
+* [Documentation, logging, testing, configuration and dev-ops](#documentation-logging-testing-configuration-and-dev-ops)
+* [Appendix: Requirements](#appendix-requirements)
+* [Appendix: Instruction for manual testing](#appendix-instruction-for-manual-testing)
+* [Non Functional Requirement (NFR)](#non-functional-requirements)
+* [Glossary](#glossary)
+___
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+___
+## Setting Up and Getting Started
+___
+## Product scope
+___
+## Target user profile
+This application is for property agent who is managing single owner rental units and is looking to reduce the overhead to filter appropriate tenants. The property agent would also want to monitor the expenses such as damages, utility bills and payment dates for rent.
+___
+## Value proposition
+Aids property agent in tracking information related to their property which includes:
+- Filtering appropriate tenants (Based on tags e.g.: gender, occupations and age)
+- Show expenses from the rented unit
+- Monitor payment dates 
 
-## Design & implementation
+Some of the constraint includes:
+- Single owner unit (Shared ownership will be registered under one owner's name)
+- Unable to calculate tax payment
+___
+## User Stories
```

Comment:
The table is not shown correctly
<img width="973" alt="Screenshot 2022-10-27 at 5 23 34 PM" src="https://user-images.githubusercontent.com/30253872/198246535-0a410921-c92e-457d-aa9d-f67ed05f31b7.png">